thrift/src/main/java/com/intellij/plugins/thrift/editor/GoToThriftDefinitionMarkerProvider.java

```diff
@@ -46,35 +57,83 @@ private boolean isThriftStruct(PsiClass element) {
   @Override
   protected void collectNavigationMarkers(@NotNull PsiElement element,
                                           @NotNull Collection<? super RelatedItemLineMarkerInfo<?>> result) {
-    if (element instanceof PsiClass && isThriftStruct((PsiClass)element)) {
+    List<ThriftDeclaration> thrift = findThriftDeclaration(element);
+    mark(element, result, thrift);
+  }
+
+  private List<ThriftDeclaration> findThriftDeclaration(@NotNull PsiElement element) {
+    if (element instanceof PsiClass && isThriftStruct((PsiClass) element)) {
       PsiClass psiClass = (PsiClass) element;
-      String name = psiClass.getName();
-      if (name != null) {
-        Stopwatch stopWatch = Stopwatch.createStarted();
-        List<ThriftDeclaration> thriftDefinitions =
-            ThriftDeclarationIndex.findDeclaration(name, psiClass.getProject(),
-                GlobalSearchScope.allScope(psiClass.getProject()));
-        Duration time = stopWatch.elapsed();
-        if (time.toMillis() > 20) {
-          logger.warn("GoToThriftDefinition action update took " + time.toMillis() + "ms");
-        }
-        if (!thriftDefinitions.isEmpty()) {
-          NavigationGutterIconBuilder<PsiElement> builder =
-              NavigationGutterIconBuilder.create(ThriftIcons.STRUCT)
-                  .setTargets(thriftDefinitions)
-                  .setTooltipText(ThriftBundle.message("thrift.goto.source"));
-          Optional<PsiElement> identifier =
-              Arrays.stream(element.getChildren()).filter(x -> x instanceof PsiIdentifier).findFirst();
-          // Some languages (Scala) don't put PsiIdentifier here, PsiModifierList is a fallback
-          Optional<PsiElement> modifiersList =
-              Arrays.stream(element.getChildren()).filter(x -> x instanceof PsiModifierList).findFirst();
-          if (identifier.isPresent()) {
-            result.add(builder.createLineMarkerInfo(identifier.get()));
-          } else if (modifiersList.isPresent()) {
-            result.add(builder.createLineMarkerInfo(modifiersList.get()));
+      return findByClass(psiClass);
```

Comment:
Extract the original logic into a separate method, and add a quick jump to ThriftDeclaration in the java method
![image](https://github.com/arhont375/intellij-thrift/assets/42882807/f9b8fc6c-578d-4635-ad85-d85705160988)
![image](https://github.com/arhont375/intellij-thrift/assets/42882807/faa424bf-f06f-4fd4-85a2-7197477393c5)